packages/simcore-sdk/src/simcore_sdk/nodeports/_item.py

```diff
@@ -0,0 +1,86 @@
+"""This module contains an item representing a node port"""
+
+import logging
+import collections
+import datetime
+from simcore_sdk.nodeports import exceptions
+from simcore_sdk.nodeports import config
+from simcore_sdk.nodeports import filemanager
+
+_LOGGER = logging.getLogger(__name__)
+
+_DataItem = collections.namedtuple("_DataItem", config.DATA_ITEM_KEYS)
+
+class DataItem(_DataItem):
+    """This class encapsulate a Data Item and provide accessors functions"""
+    def __new__(cls, **kwargs):        
+        _LOGGER.debug("Creating new data item with %s", kwargs)
+        self = super(DataItem, cls).__new__(cls, **kwargs)
+        self.new_data_cb = None

```

Comment:
Why not simply using the constructor to add these variables (i.e. ``__init__``)?

Reply:
as it is a immutable collection. __init__ is actually empty and does nothing that is why I had to do it in the constructor.